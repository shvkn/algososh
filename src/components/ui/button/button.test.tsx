import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import { Button } from "./button"

describe("Button", () => {
  it("Should render button with Text properly", () => {
    const tree = renderer.create(<Button text={"Click me"} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render button without Text properly", () => {
    const tree = renderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render Disabled button properly", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render button with Loader properly", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should react on Click event", () => {
    const mockCallback = jest.fn();
    render(<Button text={"Click me"} onClick={mockCallback} />);

    fireEvent.click(screen.getByText(/click me/i));

    expect(mockCallback.mock.calls.length).toBe(1);
  });

  it("Shouldn't react on Click event", () => {
    const mockCallback = jest.fn();
    render(<Button text={"Click me"} onClick={mockCallback} disabled={true} />);

    fireEvent.click(screen.getByText(/click me/i));

    expect(mockCallback.mock.calls.length).toBe(0);
  });
});
