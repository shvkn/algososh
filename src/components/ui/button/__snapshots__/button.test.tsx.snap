// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render Disabled button properly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Should render button with Loader properly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Should render button with Text properly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button Should render button with icon 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Should render button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button Should render button without Text properly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
